babel.config.js

```diff
@@ -0,0 +1,47 @@
+/**
```

Comment:
I don't see anything dynamic about this configuration, is there a need to migrate to a `js` file or can we stick with the static `.babelrc` until a dynamic need drives the change?

Reply:
For some reason, `babel` wasn't picking up configs like `sourceType: 'module'` from the `.babelrc` file, but was doing so from `babel.config.js`. In addition, I'm planning on adding things like `sourceMapsAbsolute: true` for only when we're running tests, and the `istanbul` transform for when we're doing coverage. Will update this PR and ping this thread.

Reply:
Makes sense, thanks!

Reply:
@kristoferbaxter To conclude this discussion, `babel.config.js` is now dynamic based on whether we're running on Travis or not. There may be scope to make more settings dynamic in future.